pkg/network/metrics/pprof.go

```diff
@@ -18,9 +18,12 @@ func NewPprofService(cfg Config) *Service {
 	handler.HandleFunc("/debug/pprof/trace", pprof.Trace)
 
 	return &Service{
-		&http.Server{
+		Server: &http.Server{
 			Addr:    cfg.Address + ":" + cfg.Port,
 			Handler: handler,
-		}, cfg, "Pprof",
+		},
+		config:      cfg,
+		serviceType: "Pprof",
+		log:         getLogger("Pprof"),
```

Comment:
Why can't get a logger via parameters?

Reply:
We can!

Reply:
Well, we should then do that for both services, let them accept logger via `NewXXX()` parameters.